external/fv3fit/tests/reservoir/test_readout.py

```diff
@@ -156,3 +165,84 @@ def test_combined_load(tmpdir):
     np.testing.assert_array_almost_equal(
         np.concatenate([r.intercepts for r in readouts]), combined_readout.intercepts
     )
+
+
+def test_BatchLinearRegressor():
+    # y0 = 1*x0 + 2*x1 + 4*x3 + 3
+    # y1 = 1*x0 + 2*x1 + 4*x3 - 1
+    X = np.array([[1, 2, 1], [2, 3, 0]])
+    y = np.array([[12.0, 8], [11, 7]])
+
+    # use_least_squares_solve needed when overdetermined test cases
+    # result in nonsingular XT.X
+    config = BatchLinearRegressorHyperparameters(
+        l2=0, add_bias_term=True, use_least_squares_solve=True
+    )
+    lr = BatchLinearRegressor(config)
+    lr.batch_update(X, y)
+
+    coefficients, intercepts = lr.get_weights()
+    np.testing.assert_array_almost_equal(np.dot(X, coefficients) + intercepts, y)
+
+
+def test_BatchLinearRegressor_iterative_result_same_as_one_shot():
+    x_dim, y_dim, N = 25, 5, 50
+    l2 = 0.1
+
+    W_truth = np.random.rand(y_dim, x_dim + 1)
+    noise = np.random.normal(-0.1, 0.1, size=(N, y_dim))
+
+    n_batches = 5
+    X_batches = [
+        np.concatenate([np.random.rand(N, x_dim), np.ones((N, 1))], axis=1)
+        for i in range(n_batches)
+    ]
+    y_batches = [np.dot(Xb, W_truth.T) + noise for Xb in X_batches]
+
+    X_full = np.concatenate(X_batches, axis=0)
+    y_full = np.concatenate(y_batches, axis=0)
+
+    # use_least_squares_solve needed when overdetermined test cases
+    # result in nonsingular XT.X
+    config = BatchLinearRegressorHyperparameters(
+        l2=l2, add_bias_term=True, use_least_squares_solve=True
+    )
+    batch_lr = BatchLinearRegressor(config)
+    for Xb, yb in zip(X_batches, y_batches):
+        batch_lr.batch_update(Xb, yb)
+    coefficients, intercepts = batch_lr.get_weights()
+
+    full_lr = BatchLinearRegressor(config)
+    full_lr.batch_update(X_full, y_full)
+    coefficients_full, intercepts_full = batch_lr.get_weights()
```

Comment:
I think this test may be falsely passing!
```suggestion
    coefficients_full, intercepts_full = full_lr.get_weights()
```

Reply:
Whoops. This typo covered up the other bug in the regularization!